amulet/sentry.py

```diff
@@ -374,6 +378,61 @@ def check_status():
             if check_status():
                 log.info('Deployment settled in %s seconds.',
                          (datetime.now() - start).total_seconds())
+                self.discover_units()
+                return
+
+    def wait_for_statuses(self, statuses, timeout=300):
+        """
+        Wait for specific workload statuses to be set via status-set.
+
+        Note that if this is called on an environment that doesn't support
+        extended status (pre Juju 1.24), it will raise a
+        :class:`~amulet.helpers.UnsupportedError` exception.
+
+        :param dict statuses: A status, or mapping of services to a workload
```

Comment:
True, I hadn't considered that possibility, and it is supported.  I'll update the docs.